src/mscorlib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -430,24 +433,14 @@ public override void BeginCatchBlock(Type exceptionType)
 
                 // if this is a catch block the exception will be pushed on the stack and we need to update the stack info
                 UpdateStackSize(OpCodes.Nop, 1);
-            }
 
-            current.MarkCatchAddr(ILOffset, exceptionType);
+                current.MarkCatchAddr(ILOffset, exceptionType);
```

Comment:
Would it make sense to keep it after the if/else block to make this more similar to the base class implementation?